File: Eclipse/w7cpframe/src/clsPfMaterial.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


  class clsPfMaterial
    {

        public int key = 0;

        public double density = 0;        //.. density ..
        public double emod = 0;           //.. elastic Modulus ..
        public double therm = 0;          //.. coeff of thermal expansion..

        public void initialise()
        {
            key = 0;
            density = 0;
            emod = 0;
            therm = 0;
        }

        public void setValues(int materialKey, double massDensity, double ElasticModulus, double CoeffThermExpansion)
        {
            key = materialKey;
            density = massDensity;
            emod = ElasticModulus;
            therm = CoeffThermExpansion;
        }

        public String sprint()
        {
            String s; 

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%15.4f",density);
            s = s + String.format("%15.4f",emod);
            s = s + String.format("%15.4f",therm);

            return s;

        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            //WScript.Echo( s);
            sgetData(s);


            System.out.println("... fgetData");
        }


        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i,n;

            System.out.println("sgetData ...");

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }


            n = dataflds.length;
            for (i = 0; i < n; i++)
            {
                System.out.println(i + "<" + dataflds[i] + ">");
            }

            key = Integer.parseInt(dataflds[0]);
            density = Double.parseDouble(dataflds[1]);
            emod = Double.parseDouble(dataflds[2]);
            therm = Double.parseDouble(dataflds[3]);


            System.out.println("... sgetData");
        }

    } //class

File: VisualStudio/VBasic/w7cpframe/clsPfMaterial.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfMaterial

#Const isUsingExcel = 0

    Public key As Integer

    Public density As Double     '.. density ..
    Public emod As Double           '.. elastic Modulus ..
    Public therm As Double          '.. coeff of thermal expansion..}


    Sub initialise()
        density = 0
        emod = 0
        therm = 0
    End Sub

    Sub setValues(ByVal materialKey As Integer, ByVal massDensity As Double, ByVal ElasticModulus As Double, ByVal CoeffThermExpansion As Double)
        key = materialKey
        density = massDensity
        emod = ElasticModulus
        therm = CoeffThermExpansion
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                 & StrLPad(Format(density, "0.00"), 15) _
                 & StrLPad(Format(emod, "0.00E+00"), 15) _
                 & StrLPad(Format(therm, "0.0000E+00"), 15)

    End Function

    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = density
            .Offset(RecNum, 2).Value = emod
            .Offset(RecNum, 3).Value = therm
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            key = .Offset(RecNum, 0).Value
            density = .Offset(RecNum, 1).Value
            emod = .Offset(RecNum, 2).Value
            therm = .Offset(RecNum, 3).Value
        End With
    End Sub

#End If

End Class
